arkouda/dataframe.py

```diff
@@ -2045,14 +2050,11 @@ def attach(user_defined_name: str) -> DataFrame:
         # Remove duplicates caused by multiple components in Categorical or SegArray and
         # loop through
         for name in set(matches):
-            colName = name.split("_")[3]
-            if f"_{Strings.objtype}_" in name or f"_{pdarray.objtype}_" in name:
-                cols_resp = cast(str, generic_msg(cmd="attach", args={"name": name}))
-                dtype = cols_resp.split()[2]
-                if dtype == Strings.objtype:
-                    columns[colName] = Strings.from_return_msg(cols_resp)
-                else:  # pdarray
-                    columns[colName] = create_pdarray(cols_resp)
+            colName = DataFrame._parse_col_name(name, user_defined_name)[0]
+            if f"_{Strings.objtype}_" in name:
+                columns[colName] = Strings.attach(name)
+            elif f"_{pdarray.objtype}_" in name:
+                columns[colName] = attach(name)
```

Comment:
I am a little worried that people may confuse this for a generic method. I see why you are only calling attach, but I do think it'd be helpful to do something like 
```
from arkouda.pdarrayclass import attach as pd_attach
```
to indicate this is specific to pdarray. Not really a sticking point, but figured I would mention.

Reply:
Updated